src/pypeh/core/models/validation_dto.py

```diff
@@ -169,40 +171,55 @@ def from_peh(
                 for nested_expr in arg_expressions
             ]
         validation_command = getattr(expression, "validation_command", "conjunction")
-        arg_values = getattr(expression, "validation_arg_values", None)
-        source_paths = getattr(expression, "validation_subject_source_paths", None)
+
+        subject_source_paths = getattr(expression, "validation_subject_source_paths", None)
         arg_type = None
-        if source_paths:
+        observable_property_id_based_subject_source_paths = []
+        if subject_source_paths:
             arg_types = set()
             if observable_property_short_name_dict is not None:
-                for source_path in source_paths:
-                    obs_prop = observable_property_short_name_dict.get(source_path, None)
+                for source_path in subject_source_paths:
                     if source_path is None:
                         me = f"Could not find validation_subject_source_path with short_name {source_path}"
                         logger.error(me)
                         raise ValueError(me)
+                    elif source_path not in observable_property_short_name_dict.keys():
```

Comment:
the if / elif logic escapes me here. Could just be if else, could it not? Also, just do `obs_prop = observable_property_short_name_dict.get(source_path, None)`. And then raise value error if obs_prop is None